packages/app/src/lib/import.ts

```diff
@@ -425,11 +431,110 @@ export function is1Pux(file: File): boolean {
     return file.name.endsWith(".1pux");
 }
 
+async function parseBitwardenItem(
+    vaultName: string,
+    item: BitwardenItem,
+    folders: BitwardenExport["folders"]
+): Promise<VaultItem | undefined> {
+    if (!item) {
+        return;
+    }
+
+    const { parseToRowData } = await import("./bitwarden-parser");
+
+    const rowData = parseToRowData(item, [vaultName], folders);
+
+    if (!rowData) {
+        return;
+    }
+
+    const itemName = rowData.name;
+    const tags = rowData.tags.split(",");
+
+    const fields: Field[] = [];
+
+    if (rowData.username) {
+        fields.push(new Field({ name: $l("Username"), value: rowData.username, type: FieldType.Username }));
+    }
+
+    if (rowData.password) {
+        fields.push(new Field({ name: $l("Password"), value: rowData.password, type: FieldType.Password }));
+    }
+
+    if (rowData.url) {
+        fields.push(new Field({ name: $l("URL"), value: rowData.url, type: FieldType.Url }));
+    }
+
+    if (rowData.notes) {
+        fields.push(new Field({ name: $l("Notes"), value: rowData.notes, type: FieldType.Note }));
+    }
+
+    for (const extraField of rowData.extraFields) {
+        if (extraField.type === "totp") {
+            // Extract just the secret
+            try {
+                const secret = new URL(extraField.value).searchParams.get("secret");
+                if (secret) {
+                    fields.push(new Field({ name: extraField.name, value: secret, type: FieldType.Totp }));
+                }
+            } catch (error) {
+                // Do nothing
+            }
+        } else {
+            fields.push(
+                new Field({ name: extraField.name, value: extraField.value, type: extraField.type as FieldType })
+            );
+        }
+    }
+
+    return createVaultItem({ name: itemName, fields, tags });
+}
+
+export async function asBitwarden(file: File): Promise<VaultItem[]> {
+    try {
+        const { parseBitwardenFile } = await import("./bitwarden-parser");
+        const contents = await readFileAsText(file);
+        const dataExport = await parseBitwardenFile(contents);
+
+        const items = [];
+
+        for (const vaultItem of dataExport.items) {
+            if (vaultItem) {
+                const parsedItem = await parseBitwardenItem(
+                    file.name.replace(".json", ""),
+                    vaultItem,
+                    dataExport.folders
+                );
+                if (parsedItem) {
+                    items.push(parsedItem);
+                }
+            }
+        }
+
+        return items;
+    } catch (error) {
+        throw new Err(ErrorCode.INVALID_BITWARDEN, "Failed to parse Bitwarden .json file.");
+    }
+}
+
+export async function isBitwarden(file: File): Promise<boolean> {
+    try {
+        const content = await readFileAsText(file);
+        const data = JSON.parse(content);
+        return Object.prototype.hasOwnProperty.call(data, "items");
```

Comment:
Just curious: Why not simply `data.hasOwnProperty("items")` or `Boolean(data.items)`?

Reply:
Also, is there no better/more reliable name to identify a bitwarden file? It doesn't seem too unlikely that there are other export formats that might include an `items` property.

Reply:
You can read https://stackoverflow.com/questions/12017693/why-use-object-prototype-hasownproperty-callmyobj-prop-instead-of-myobj-hasow/ or https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty#using_hasownproperty_as_a_property_name for more thorough explanations, but basically, there's no guarantee `object.hasOwnProperty(property)` will work as expected (especially in client-side code), whereas `Object.prototype.hasOwnProperty.call(object, property)` will.

As for a more reliable name, there isn't, no, and if we run into problems, we can simply try to confirm there are some more required properties, like `items[x].type` and so on.

Reply:
Alright! Thanks for the explantation :)